fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessingViewModel.java

```diff
@@ -120,23 +120,43 @@ public void initialize(String laoId) {
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe(
                 witnessMessage -> {
+                  if (witnessMessage.isEmpty()) {
+                    return;
+                  }
                   // Order by latest arrived
                   setWitnessMessages(
                       witnessMessage.stream()
                           .sorted(Comparator.comparing(WitnessMessage::getTimestamp).reversed())
                           .collect(Collectors.toList()));
 
-                  // When a new witness message is received, if it needs to be signed by the user
-                  // then we show a pop up that the user can click to open the witnessing fragment
+                  WitnessMessage lastMessage =
+                      Objects.requireNonNull(witnessMessages.getValue()).get(0);
+
+                  // When a new witness message is received, if it needs to be yet signed by the
+                  // user
+                  // then we show a pop up that the user can click to open the witnessing fragment.
+                  // Don't show the pop-up for the organizer as we use an automatic signature
+                  // mechanism
                   PublicKey myPk = keyManager.getMainPublicKey();
-                  if (!witnessMessage.isEmpty()
-                      && witnessingRepo.isWitness(laoId, myPk)
-                      && !Objects.requireNonNull(witnessMessages.getValue())
-                          .get(0)
-                          .getWitnesses()
-                          .contains(myPk)) {
-                    showPopup.setValue(true);
+                  boolean isOrganizer = lao.isOrganizer(keyManager.getMainPublicKey());
+                  boolean alreadySigned = lastMessage.getWitnesses().contains(myPk);
+                  if (isOrganizer && !alreadySigned) {
+                    // Automatically sign the messages if it's the organizer
+                    disposables.add(
+                        signMessage(lastMessage)
+                            .subscribe(
+                                () ->
+                                    Timber.tag(TAG)
```

Comment:
random question: I see we're using Timber now.
Does the CI include the timber-lint tool to ensure that Timber is used properly ? 🤔

Reply:
No, it doesn't. Timber was a "new" entry of last semester that I implemented to have the remote logging functionality. But it would be a nice add-on to have in the CI